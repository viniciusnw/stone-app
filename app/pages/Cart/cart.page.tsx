import React from "react";
import { View, Text, ActivityIndicator } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { Button, Icon, Logo } from "@Components";

const Cart = (props: any) => {
  const navigation = useNavigation();

  React.useEffect(
    () =>
      props.setState({
        bottomBar: {
          onPress: () => console.log("onNextBottomPress"),
          text: `Finalizar`,
          label: `Total: 299,90`,
          description: `ou 3x R$100`,
        },
        topBar: {
          title: "Cart",
          back: navigation.goBack,
        },
      }),
    []
  );

  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <Text />
    </View>
  );
}

export default Cart;