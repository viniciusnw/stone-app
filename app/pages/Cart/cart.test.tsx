import React from "react";
import renderer from "react-test-renderer";
import { lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import Cart from "./cart.page";
import "jest-styled-components";
import Wrapper from "@/app/navigation/wrapper.navigator";

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
    useRoute: () => ({
      params: {
        productId: 1,
      },
    }),
  };
});

test("CartWrapper", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <Wrapper Page={Cart} header bottomBarType={"action-bar"} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Cart", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <Cart setState={jest.fn} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
