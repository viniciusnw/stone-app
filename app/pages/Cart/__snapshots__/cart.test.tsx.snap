// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
</View>
`;

exports[`CartWrapper 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ddd",
        "borderColor": "transparent",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "width": 30,
        }
      }
    />
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#111111",
          "fontSize": 20,
          "fontWeight": "700",
          "maxWidth": 200,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "FredokaOne-Regular",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0,168,104)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Stone
          </Text>
          <Text
            style={
              {
                "color": "rgb(10,245,61)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Ton :)
          </Text>
        </Text>
      </View>
    </Text>
    <View
      style={
        {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "height": 62,
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#666666",
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 14,
              "fontWeight": "700",
            }
          }
        >
          <Text>
            ...
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
